ui-v2/src/api/flow-runs/use-paginate-flow-runs-with-flows/use-paginate-flow-runs-with-flows.ts

```diff
@@ -0,0 +1,82 @@
+import {
+	type FlowRunsPaginateFilter,
+	buildPaginateFlowRunsQuery,
+} from "@/api/flow-runs";
+import { Flow, buildListFlowsQuery } from "@/api/flows";
+import { useQuery } from "@tanstack/react-query";
+import { useMemo } from "react";
+
+/**
+ *
+ * @param filter
+ * @returns a simplified query object that joins a flow run's pagination data with it's parent flow
+ */
+export const usePaginateFlowRunswithFlows = (
+	filter: FlowRunsPaginateFilter,
+) => {
+	const { data: paginateFlowRunsData, error: paginateFlowRunsError } = useQuery(
+		buildPaginateFlowRunsQuery(filter),
+	);
+
+	const flowIds = useMemo(() => {
+		if (!paginateFlowRunsData) {
+			return new Set<string>();
+		}
+		const flowIds = paginateFlowRunsData.results.map(
+			(flowRun) => flowRun.flow_id,
+		);
+		return new Set<string>(flowIds);
```

Comment:
nit: There's probably no benefit to creating a set here. There shouldn't be any repeated flow run ids in the results of the flow run data. So creating the set just to turn it back into an array isn't necessary.